libmamba/src/core/util.cpp

```diff
@@ -848,127 +889,23 @@ namespace mamba
         return ret;
     }
 
-    bool LockFile::unlock()
+    bool LockFileOwner::unlock()
     {
         int ret = 0;
 
         // POSIX systems automatically remove locks when closing any file
         // descriptor related to the file
 #ifdef _WIN32
-        LOG_TRACE << "Removing lock on '" << m_lock.string() << "'";
+        LOG_TRACE << "Removing lock on '" << m_lockfile_path.string() << "'";
         _lseek(m_fd, MAMBA_LOCK_POS, SEEK_SET);
         ret = _locking(m_fd, LK_UNLCK, 1 /*lock_file_contents_length()*/);
 #endif
         remove_lockfile();
```

Comment:
Hi,
This may just be "shouting from the sidelines", so take this with a heavy grain of salt 😉 

Removing the lock file seems a bit odd here. We ran into a similar problem in our development a while back.
Having the lock file in place enabled us to see what is locking it, and removing the file kind of made this hidden and caused some extra hassle.
..but if you can provide some test build we can test this change with our RCC builds (https://github.com/robocorp/rcc)

Reply:
Hi, thanks for pointing this. However I'm not sure what you find odd exactly as this is the `unlock` function called when the lock have to be released, so removing the lockfile is kind of mandatory? Or did I misunderstand something?
Also I removed the pid written in the lockfile, do you think it should be kept for debugging purpose? We don't need it anymore here but it's not clear to me if it's still useful for users.

As for the build I'm not sure how I can provide it based on the CI output yet but I'm kind of new around here, I'll ask around and let you know.

Reply:
Hi, 
As said, I might be way off base, but the way we ended up doing the file locks (a simplification as I was not the one implementing 😉) is just to have the file there, take the lock from the process, and ask the OS if the file has locks or not. 

That way, the lock file always exists, and in debugging cases you can go in and check with other tools also what is holding the lock.

Reply:
Thanks, I'll discuss about it with the others, I'll consider making a PR with that logic. We have plans to refactor/review/revamp this lockfile system anyway so that's good input. 👍🏽